clcache.py

```diff
@@ -164,33 +164,43 @@ def clean(self, stats, maximumSize):
         except ImportError:
             walker = os.walk
 
-        objects = [os.path.join(root, "object")
-                   for root, folder, files in walker(self.objectsDir)
-                   if "object" in files]
+        objectInfos = []
+        for x in xrange(16):
+            try:
+                objects = [os.path.join(root, "object")
+                       for root, folder, files in walker(self.objectsDir)
+                       if "object" in files]
+
+                objectInfos = [(os.stat(fn), fn) for fn in objects]
```

Comment:
no matter if there is a reason to manually delete the cache or not ... people will do it :/

this patch won't add a lot of overhead, but will make the code more robust.